dataloader/dataloader.hpp

```diff
@@ -0,0 +1,175 @@
+   * @param useScaler Fits the scaler on training data and transforms dataset.
+   * @param dropHeader Drops the first row from CSV.
+   * @param startInputFeatures First Index which will be fed into the model as input.
+   *                           Note: Indicies are wrapped and -1 implies last
+   *                           column.
+   * @param endInputFeature Last Index which will be fed into the model as input.
+   *                        Note: Indicies are wrapped and -1 implies last
+   *                        column.
+   * @param startPredictionFeatures First Index which be predicted by the model as output.
+   *                                Note: Indicies are wrapped and -1 implies last
+   *                                column.
+   * @param endPredictionFeatures Last Index which be predicted by the model as output.
+   *                              Note: Indicies are wrapped and -1 implies last
+   *                              column.
+   * @param augmentation Vector strings of augmentations supported by mlpack.
+   * @param augmentationProbability Probability of applying augmentation to a particular cell.
+   */
+  void LoadCSV(const std::string &datasetPath,
+               const bool loadTrainData = true,
+               const bool shuffle = true,
+               const double ratio = 0.75,
+               const bool useScaler = false,
+               const bool dropHeader = false,
+               const int startInputFeatures = -1,
+               const int endInputFeatures = -1,
+               const int startPredictionFeatures = -1,
+               const int endPredictionFeatures = -1,
+               const std::vector<std::string> augmentation =
+                   std::vector<std::string>(),
+               const double augmentationProbability = 0.2);
+
+  //! Get the Training Dataset.
+  DataSetX TrainX() const { return trainX; }
+
+  //! Modify the Training Dataset.
+  DataSetX &TrainX() { return trainX; }
+
+  //! Get the Training Dataset.
+  DataSetY TrainY() const { return trainY; }
+  //! Modify the Training Dataset.
+  DataSetY &TrainY() { return trainY; }
+
+  //! Get the Test Dataset.
+  DataSetX TestX() const { return testX; }
+  //! Modify the Test Dataset.
+  DataSetX &TestX() { return testX; }
+
+  //! Get the Test Dataset.
+  DataSetY TestY() const { return testY; }
+  //! Modify the Training Dataset.
+  DataSetY &TestY() { return testY; }
+
+  //! Get the Validation Dataset.
+  DataSetX ValidX() const { return validX; }
+  //! Modify the Validation Dataset.
+  DataSetX &ValidX() { return validX; }
+
+  //! Get the Validation Dataset.
+  DataSetY ValidY() const { return validY; }
+  //! Modify the Validation Dataset.
+  DataSetY &ValidY() { return validY; }
+
+  //!Get the Scaler.
+  ScalerType Scaler() const { return scaler; }
+  //! Modify the Sclaer.
+  ScalerType &Scaler() { return scaler; }
+
+private:
+  // Utility Function to wrap indices.
+  size_t wrapIndex(int index, size_t length)
```

Comment:
```suggestion
  size_t WrapIndex(int index, size_t length)
```
(Just to match the mlpack style.  I didn't adapt usages of this function throughout the code though.)